### PR TITLE
Obsolete older -doc packages in main package to prevent -doc downgrades

### PR DESCRIPTION
**Pull request by shundhammer**
## Bugzilla

https://bugzilla.suse.com/show_bug.cgi?id=1184363

## Problem

We dropped the libyui*-doc packages recently. But when a user still has any of them installed and upgrades to SLE-15 SP3, he will get older outdated versions of those packages installed since there are no more recent ones.

## Fix

Each of the main packages (the one containing the libyui binaries) now obsoletes all -doc packages with an older version than itself, so they will be uninstalled upon upgrade.

Should we ever decide to reintroduce those -doc packages, that `<` comparison should make it easy to do it on the fly (without even removing that `obsoletes` dependency).

## Target Branch

This is for the SLE-15-SP3 branch. We will only merge this if the PM agrees.

**Update** 2021-04-07 10:45:  sweiberg writes that yes, he wants it in SLE-15 SP3.

This will also go to master in a later PR.

### CODE REVIEW

**Review by mvidner — APPROVED** [review]
LGTM

I have also double checked the package list, with

```console
$ zypper se -t package libyui*doc
S | Name                   | Summary                          | Type
--+------------------------+----------------------------------+--------
  | libyui-doc             | Libyui documentation             | package
  | libyui-ncurses-doc     | Libyui-ncurses documentation     | package
  | libyui-ncurses-pkg-doc | Libyui-ncurses-pkg documentation | package
  | libyui-qt-doc          | Libyui-qt documentation          | package
  | libyui-qt-graph-doc    | Libyui-qt-graph documentation    | package
  | libyui-qt-pkg-doc      | Libyui-qt-pkg documentation      | package
```

**Review by joseivanlopez — APPROVED** [review]
LGTM

**Review by mvidner — CHANGES_REQUESTED** [review]
CI:
> ci-tasks: line 44: syntax error near unexpected token `<'

That's because the script needs `bash` but ci.yml runs it via `sh`

File: ci-tasks [hunk 1]
```diff
@@ -6,7 +6,9 @@ set -e -x
 rake package
 
 # run the osc source validator to check the .spec and .changes locally
-(cd package && /usr/lib/obs/service/source_validator)
+# Workaround for CI failure:
+# Don't validate on SLE-15-SP3 since that image has an outdated validator
```
Comment (mvidner) [inline ci-tasks:10]: I'd like to understand the problem before disabling the check. Any pointers?

Comment (shundhammer) [reply ci-tasks:10]: It had complained "you can have either multibuild or multiple .spec files, but not both". @joseivanlopez knows more.

**Comment by shundhammer** [comment]
> CI:
> 
> > ci-tasks: line 44: syntax error near unexpected token `<'
> 
> That's because the script needs `bash` but ci.yml runs it via `sh`

Yes, that's what I suspected yesterday. So it looks like we need to simplify that even for more basic shells.

**Comment by mvidner** [comment]
The rationale for `mapfile` is this: https://github.com/koalaman/shellcheck/wiki/SC2207

So, is the problem that Bash is not available? If Bash is there we should simply set the executable bit and call it directly.

**Comment by shundhammer** [comment]
Or simply do it in a more pedestrian way that will probably also be easier to understand. It might require a loop and more steps. And we might consider putting it into a function with a speaking name. Right now it's difficult for anyone reading the code what exactly this line does - and why.

File: ci-tasks (hunk 1 above)
Comment (lslezak) [reply ci-tasks]: In the SP3 builds we use Leap 15.3 based image which contains an old source validator which does not properly support multibuild packages.

In `master` (TW based image) with a newer version it works properly.

**Review by mvidner — APPROVED** [review]
I have re-enabled the OBS validator and fixed the GH task to run ci-tasks directly

**Pull request merged**